### ISSUE TITLE
Support for Ceph radosgw with S3 adapter

### ISSUE THREAD

**Issue by discotroy** [thephpleague/flysystem#662]
Hybrid bug report / plea for assistance!

Ceph provides an S3-compatible object storage service, radosgw.  I am attempting to get a Laravel-based application to store files in this service via Flysytem and the AwsS3 interface but failing.

I've tried using both v2 and v3 of the AWS SDK, but have had no success.  v3 fails consistently as radosgw doesn't appear to support the AWS4 signature method and I don't know how to force it to use the old-style signature.  v2 gets slightly further, but fails consistently with a "403 / signature not valid" error.

I have tested the service with both Boto and with s3cmd to ensure that it is at least vaguely compliant, and both work.  From what I can tell, the difference between what s3cmd sends and what Flysystem sends is a header ("Accept-Encoding: identity", present with s3cmd but not Flysystem).  It might be worth noting from observing packet captures between s3cmd and the service, that s3cmd initially attempts the AWS4 signaure, detects a failure and falls back to old-style.

I've spent two days already debugging this and trying to work out what needs to be amended to support Radosgw, but I'm about at my wits end.  All I can think of is that it is something relating to the signature generation perhaps.

**Comment by discotroy** [comment]
Disregard.  It turns out that it was the authorisation hash that was causing the problem, but it was down to my not setting up bucket-based / vhost URLs and persisting in using path-based URLs.  I went back to basics and wrote a test using only the AWS SDKv2, which worked.  Then built up to using Flysystem's layer, again using AWS SDKv2, which worked.  I then attempted to use the Laravel Flysystem bridge, but this seems to require AWS SDKv3, which I cannot use.

Three days later, I'm a little wiser.
